### PR TITLE
Add reduce option to F.contrastive

### PR DESCRIPTION
**Pull request by delta2323**
This PR adds `reduce` option to `F.contrastive`. I also fix its document. Related to #2558

### CODE REVIEW

File: chainer/functions/loss/contrastive.py [hunk 1]
```diff
@@ -39,9 +45,9 @@ def forward(self, inputs):
         self.dist = xp.sqrt(self.dist_sq)
         self.mdist = self.margin - self.dist
         dist = xp.maximum(self.mdist, 0)
-        loss = y * self.dist_sq + (1 - y) * dist * dist
-        loss = xp.sum(loss) / 2.0 / x0.shape[0]
-
+        loss = (y * self.dist_sq + (1 - y) * dist * dist) / 2.0
```
Comment (unnonouno) [inline chainer/functions/loss/contrastive.py]: `* 0.5` is better.

File: tests/chainer_tests/functions_tests/loss_tests/test_contrastive.py [hunk 2]
```diff
@@ -65,30 +77,53 @@ def test_forward_gpu_no_cudnn(self):
...
     def test_backward_cpu(self):
-        self.check_backward(self.x0, self.x1, self.t)
+        self.check_backward(self.x0, self.x1, self.t, self.gy)
 
     @attr.gpu
     @condition.retry(3)
     def test_backward_gpu_no_cudnn(self):
         self.check_backward(cuda.to_gpu(self.x0), cuda.to_gpu(self.x1),
-                            cuda.to_gpu(self.t))
+                            cuda.to_gpu(self.t), cuda.to_gpu(self.gy))
```
Comment (unnonouno) [inline tests/chainer_tests/functions_tests/loss_tests/test_contrastive.py]: When `recuce` is `'no'`,  `self.gy` is `None`

**Comment by unnonouno** [comment]
LGTM
